devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -53,24 +72,57 @@ metadata {
-        fingerprint mfr: "0312", prod: "AC01", model: "4001", deviceJoinName: "New One Outlet"  // N4001 New One  Mini Smart Plug
+        fingerprint mfr: "0312", prod: "C000", model: "C009", deviceJoinName: "Minoston Outlet", ocfDeviceType: "oic.d.smartplug" // old MP21Z
+        fingerprint mfr: "0312", prod: "FF00", model: "FF0C", deviceJoinName: "Minoston Outlet", ocfDeviceType: "oic.d.smartplug" //MP21Z Minoston Mini Smart Plug
+        fingerprint mfr: "0312", prod: "AC01", model: "4001", deviceJoinName: "New One Outlet",  ocfDeviceType: "oic.d.smartplug" // N4001 New One  Mini Smart Plug
+        fingerprint mfr: "0312", prod: "EE00", model: "EE01", deviceJoinName: "Minoston Switch", ocfDeviceType: "oic.d.switch"    //MS10ZS Minoston Smart Switch
+        fingerprint mfr: "0312", prod: "EE00", model: "EE03", deviceJoinName: "Minoston Switch", ocfDeviceType: "oic.d.switch"    //MS12ZS Minoston Smart on/off Toggle Switch
+        fingerprint mfr: "0312", prod: "A000", model: "A005", deviceJoinName: "Evalogik Switch", ocfDeviceType: "oic.d.switch"    //ZW30
+        fingerprint mfr: "0312", prod: "BB00", model: "BB01", deviceJoinName: "Evalogik Switch", ocfDeviceType: "oic.d.switch"    //ZW30S Evalogik Smart on/off Switch
+        fingerprint mfr: "0312", prod: "BB00", model: "BB03", deviceJoinName: "Evalogik Switch", ocfDeviceType: "oic.d.switch"    //ZW30TS Evalogik Smart on/off Toggle Switch
     }
 
     preferences {
-        configParams.each {
-            if (it.name) {
-                if (it.range) {
-                    input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-                } else {
-                    input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
-                }
-            }
-        }
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Paddle Control'is only valid for the devices with product number of MS10ZS, MS12ZS, ZW30, ZW30S, ZW30TS(one of them)",
+                required: false
+        getConfigParamInput(paddleControlParam)
     }
 }
 
+private getConfigParamInput(param) {
+    if (param.range) {
+        input "configParam${param.num}", "number", title: "${param.name}:", required: false, defaultValue: "${param.value}", range: param.range
+    } else {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
+    }
+}
+
+private addChildButton() {
+    log.warn "Creating Button Device"
+    def child = addChildDevice(
+            "smartthings",
```

Comment:
@sky-nie Did You double-check (with a real device) that this child device ("smartthings" namespace) is working correctly with Your DTH ("sky-nie" namespace) ?

Reply:
Yes, I have checked a moment ago, when I set the PaddleControlOptions as "Toggle", The child button can receive "pushed", And I can set the action for child button.

Reply:
But our project manager said that only "pushed" action associations can be realized, obviously there are not as many as expected (the original planned action associations are "pushed", "down", "down_2x", "up", "up_2x" "), since Automation has been enhanced in the SmarthThings App, the functions related to CentralScene can be deleted.

Reply:
But our project manager said that only "pushed" action associations can be realized, obviously there are not as many as expected (the original planned action associations are "pushed", "down", "down_2x", "up", "up_2x" "), since Automation has been enhanced in the SmarthThings App, the functions related to CentralScene can be deleted.